### PR TITLE
fix(tekton): Git org and repo should be replaced consistently

### LINKED ISSUE #4432: Mismatch in REPLACE_ME_APP_NAME during import and pipeline generation
### Summary 
There is an issue with **jx step syntax effective**. 

During **jx import** the folder name inside **charts** and **REPLACE_ME_APP_NAME** is replaced by **kube.ToValidName(strings.ToLower(repo_name))**. 

But during **jx step syntax effective** it's replaced by just raw repo_name causing inconsistencies

### Steps to reproduce the behavior
Create a repo with name a_b_c
Run **jx import --dry-run** and check charts folder app name
Run **jx step syntax effective** and check dir
There is a mismatch

### Expected behavior
Both names are same

### Actual behavior
Names are different

### Jx version

The output of `jx version` is:
```
NAME               VERSION
jx                 2.0.329
Kubernetes cluster v1.11.8-eks-7c34c0
kubectl            v1.14.0
helm client        v2.11.0+g2e55dbe
helm server        v2.14.0+g05811b8
git                git version 2.13.1
Operating System   Mac OS X 10.14.4 build 18E227
```
### Jenkins type

- [x] Next Generation (Tekton + Prow)
- [ ] Classic Jenkins
- [ ] Serverless Jenkins (JenkinsFileRunner + Prow)

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?
EKS

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

`import` does the same replacements as `pkg/jenkinsfile/pipeline.go` does, but for Git org and repo, the replacement goes through `kube.ToValidName(strings.ToLower(...))`. We should be doing that in `pipeline.go` for build pack transformation as well.

And while I was here, I moved `pkg/kube/names.go` to `pkg/util/names.go` and renamed the functions to `ToValidKubeName` (and so on), since the `kube` package depends on way too many other packages as is (so I couldn't use it in the `jenkinsfile` package, for example).

#### Special notes for the reviewer(s)

/assign @dwnusbaum 
/assign @hferentschik 

#### Which issue this PR fixes

fixes #4432

### CODE REVIEW

File: pkg/util/names.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package kube
+package util
```
Comment (hferentschik) [inline pkg/util/names.go]: OOI, what is the motivation to move package? 

On that note, there is a k8s.go in the util package. It contains an `EncodeKubernetesName` which does not even seem to be used. What's about moving all the functions in this file into there (potentially removing `EncodeKubernetesName`) on the go?

names.go is very generic and when just looking at file names in a package. I somehow find `k8s.go` gives me a better idea on what to expect, leaving potentially the door open for none name related utility functions.

Comment (abayer) [reply pkg/util/names.go]: Circular dependency hell, basically. That said, I'll move it into `k8s.go`.

**Comment by abayer** [comment]
Fair - will fix.

File: pkg/util/names.go (hunk 1 above)
Comment (hferentschik) [reply pkg/util/names.go]: I see.

**Comment by abayer** [comment]
@pmuir - done (and resolved a merge conflict when I noticed it)

**Comment by hferentschik** [comment]
> Moving this out of the kube package and into util is not right as util is bloated enough.

Same as kube really. I think ideally it would stay in `kube` and we find another way to resolve the circular dependency.

> If you can't solve the circular dependencies then suggest you move it a new package like kubeutil nested under kube.

I'd argue `util` is in this case better package. Mainly also in the hope that more people come across it. My thinking is that people are more likely to "explore" `util` at some state in order to find useful functions than any other package. Also I struggle with `kube` vs `kubeutil`. Last but not least, there is k8s.go which, in the case of a `kubeutil` package, should be moved there.

> It would also be sensible to do this as two commits, one showing the package change and one showing the actual functional changes you are making

+1

**Comment by hferentschik** [comment]
@abayer, ooi do you still have the compile error of the circular dependency?

**Comment by abayer** [comment]
@hferentschik:

```
import cycle not allowed
package main
	imports github.com/jenkins-x/jx/cmd/jx/app
	imports github.com/jenkins-x/jx/pkg/cmd
	imports github.com/jenkins-x/jx/pkg/cmd/add
	imports github.com/jenkins-x/jx/pkg/apps
	imports github.com/jenkins-x/jx/pkg/environments
	imports github.com/jenkins-x/jx/pkg/gits
	imports github.com/jenkins-x/jx/pkg/auth
	imports github.com/jenkins-x/jx/pkg/vault
	imports github.com/jenkins-x/jx/pkg/config
	imports github.com/jenkins-x/jx/pkg/jenkinsfile
	imports github.com/jenkins-x/jx/pkg/kube
	imports github.com/jenkins-x/jx/pkg/jenkinsfile
```

**Comment by abayer** [comment]
Specifically it's that `ModifyNamespace` in `kube/env.go` does stuff with `PipelineConfig`.

**Comment by hferentschik** [comment]
> Specifically it's that ModifyNamespace in kube/env.go does stuff with PipelineConfig.

Maybe breaking this cycle would be better. Creating an _kube/env_ package and moving env.go in there might work. _env_cache.go_ could move as well in this case together with the relevant tests. That's what I would try, but I leave it up to you guys.

**Comment by abayer** [comment]
@pmuir - your call - if you'd prefer I break the cycle, it looks like `env.go` (and related tests) could be pulled into their own package under `kube` without breaking anything.

**Comment by abayer** [comment]
More specifically, `env.go`, `env_modify_namespace_integration_test.go`, `env_test.go`, `env_cache.go`, `envrolebindings.go`, and `envrolebindings_test.go` can all move - `envvar.go` can't, though.

**Comment by pmuir** [comment]
I agree, the best bet is to break the cycle. @hferentschik understood re naming/packages, otoh there is sooo much in util, it's very hard to know what it does!

If we can use this as an opportunity to clean up `kube`, then I'm definitely for it - that's the next big problem after `cmd`.

**Comment by abayer** [comment]
Alrightie, change incoming momentarily!

**Comment by abayer** [comment]
...ok, not momentarily, because I think there are two more circular deps. Lemme unravel those.

**Comment by abayer** [comment]
ha, and the things I need to move to break the circular dep themselves depend on stuff from various other things in `kube`...most notably the stuff in `names.go`, but a bunch of other files as well. @pmuir, @hferentschik - for now, I'm just going to move `names.go` into `pkg/kube/naming` and leave it at that, if that's ok. I'll open another issue for reorganizing `pkg/kube` as a whole.

**Comment by pmuir** [comment]
Sounds good to me.

**Comment by abayer** [comment]
Opened https://github.com/jenkins-x/jx/issues/4474

**Comment by abayer** [comment]
whoops, missed `pipeline.go` in the move for some reason. Sigh. Fixed.

**Comment by abayer** [comment]
...no, my IDE didn't actually fix things for me on the rename. Sigh. Fixing everything else now.

**Comment by abayer** [comment]
Yay, it's good! Whenever you're ready, @pmuir or @hferentschik

**Comment by hferentschik** [comment]
I like `naming` better than `kubeutils`. Thanks for opening  #4474

/approve

**Comment by abayer** [comment]
Still looking for an lgtm... =)

**Comment by abayer** [comment]
/hold

Putting on hold until #4489 gets through so we can actually have a functional release build.

**Pull request merged**